core/blockchain.go

```diff
@@ -1822,6 +1824,21 @@ func (bc *BlockChain) insertSideChain(block *types.Block, it *insertIterator) (i
 	if parent == nil {
 		return it.index, nil, nil, errors.New("missing parent")
 	}
+	if parent.Number.Cmp(common.Big1) <= 0 {
```

Comment:
Problem: on a private network with PoA, sealers will sometimes need to reorg from block `0`.

Reply:
Fix: also check that the common ancestor is not furhter away than whatever we use for reorg size, 3 epochs or something